lib/protocol/protocol.js

```diff
@@ -332,9 +377,103 @@ function decodeMetadataResponse (resp) {
 
   if (!_.isEmpty(errors)) out.error = errors;
   out.metadata = topics;
+
+  if (version > 0) {
+    out.clusterMetadata = {
+      controllerId
+    };
+  }
+
   return [brokers, out];
 }
 
+function encodeCreateTopicRequest (clientId, correlationId, topics, timeoutMs) {
+  var request = encodeRequestHeader(clientId, correlationId, REQUEST_TYPE.createTopics, 1);
+  request.Int32BE(topics.length);
+  topics.forEach(function (topic) {
+    request.Int16BE(topic.topic.length).string(topic.topic);
+    request.Int32BE(topic.partitions);
+    request.Int16BE(topic.replicationFactor);
+    request.Int32BE(0);
+    request.Int32BE(0);
+  });
+  request.Int32BE(timeoutMs);
+  request.Int8(0);
+
+  return encodeRequestWithLength(request.make());
+}
+
+function decodeCreateTopicResponse (resp) {
+  var errors = [];
+
+  Binary.parse(resp)
+    .word32bs('size')
+    .word32bs('correlationId')
+    .word32bs('topicNum')
+    .loop(decodeTopics);
+
+  function decodeTopics (end, vars) {
+    if (vars.topicNum-- === 0) return end();
+
+    this.word16bs('topic')
+      .tap(function (vars) {
+        this.buffer('topic', vars.topic);
+        vars.topic = vars.topic.toString();
+      })
+      .word16bs('errorCode')
+      .word16bs('errorMessage')
+      .tap(function (vars) {
+        if (vars.errorCode === 0) {
+          return;
+        }
+
+        const TIMEOUT_ERROR_CODE = 7;
+        if (vars.errorCode === TIMEOUT_ERROR_CODE) {
```

Comment:
We should use the error map in the `ERROR_CODE` or just use the `createGroupError` function.